janus.c

```diff
@@ -4643,6 +4643,7 @@ gint main(int argc, char *argv[])
 	char *turn_rest_api = NULL, *turn_rest_api_key = NULL;
 #ifdef HAVE_TURNRESTAPI
 	char *turn_rest_api_method = NULL;
+	uint turn_rest_api_timeout = 0;
```

Comment:
It looks like if `turn_rest_api_timeout` is not set in the config file, the above `0` value is passed to `janus_ice_set_turn_rest_api`, which then overwrites `api_timeout` in`turnrest.c` thus indeed causing the indefinite wait you wanted to prevent.